packages/dapp/src/components/EndorsementView/Overview.tsx

```diff
@@ -0,0 +1,124 @@
+'use client';
+import React, { useEffect, useState } from 'react';
+import { ProfileAvatar } from '../ProfileAvatar';
+import { TwitterIcon, TwitterShareButton } from 'react-share';
+import { startsWithVowel } from '@/utils';
+import Link from 'next/link';
+import { MemoizedImage } from '../MemoizedImage';
+import { Input } from '../ui/input';
+import { Button } from '../ui/button';
+
+type OverviewProps = {
+  endorser: string;
+  endorsementType: string;
+  endorsee: string;
+  endorseeAvatar: string;
+  comment: string;
+  uid: string;
+  endorserAvatar: any;
+};
+
+export const Overview = ({
+  endorsementType,
+  endorsee,
+  endorseeAvatar,
+  comment,
+  endorserAvatar,
+  uid,
+}: OverviewProps) => {
+  const [pathname, setPathname] = useState<string>('');
+
+  useEffect(() => {
```

Comment:
Is this really necessary ?

Reply:
Error is thrown when accessing `window.location` if component is not properly loaded